source/hands_on/gtrxl.rst

```diff
@@ -0,0 +1,139 @@
+GTrXL
+^^^^^^^
+
+Overview
+---------
+Gated Transformer-XL, or GTrXL, first proposed in `Stabilizing Transformers for Reinforcement Learning <https://arxiv.org/pdf/1910.06764.pdf>`_,
+is a novel framework for reinforcement learning based on the Transformer-XL architecture.
+It introduces several architectural modifications that improve the stability and learning speed of Transformer including:
+placing the layer normalization on only the input stream of the submodules, and replacing residual connections with gating layers.
+The proposed architecture, surpasses LSTMs on challenging memory environments and achieves state-of-the-art
+results on the several memory benchmarks, exceeding the performance of an external memory architecture.
+
+Quick Facts
+-------------
+1. GTrXL can serve as a **backbone** for many RL algorithms.
+
+2. GTrXL only supports **sequential** observations.
+
+3. GTrXL is based on **Transformer-XL** with **Gating connections**.
+
+4. The DI-engine implementation of GTrXL is based on the R2D2 algorithm. In the original paper, it is based on the algorithm V-MPO.
+
+Key Equations or Key Graphs
+---------------------------
+**Transformer-XL**: is a Transformer architecture that introduces the notion of recurrence to the deep self-attention network.
+Instead of computing the hidden states from scratch for each new segment, Transformer-XL reuses the hidden states obtained in previous segments.
+The reused hidden states serve as memory for the current segment, which builds up a recurrent connection between the segments.
+As a result, modeling very long-term dependency becomes possible because information can be propagated through the recurrent connections.
+As an additional contribution, the Transformer-XL uses a new relative positional encoding formulation that generalizes to attention lengths longer than the one observed during training.
+
+.. image:: images/transformerXL_train_eval.png
+   :align: center
+
+**Identity map reordering**: move the layer normalization to the input stream of the submodules.
+A key benefit to this reordering is that it now enables an identity map from the input of the transformer at the first layer to the output of the transformer after the last layer.
+This is in contrast to the canonical transformer, where there are a series of layer normalization operations that non-linearly transform the state encoding.
+One hypothesis as to why the Identity Map Reordering improves results is as follows: assuming that the submodules at initialization produce values that are in expectation near
+zero, the state encoding is passed un-transformed to the policy and value heads, enabling the agent to learn a Markovian policy at the start of training.
```

Comment:
Why when the submodules at initialization produce values that are in expectation near
zero, then the state encoding is passed un-transformed to the policy and value heads and why in this case, the agent learn a Markovian policy? I don't quite understand. Could you explain more thoroughly?

Reply:
![image](https://user-images.githubusercontent.com/41103541/160961676-73efdb50-6d7e-4538-b195-98cb4f10d7d3.png)
You have to look at this figure to understand: if the output of the attention block is near 0, the input of the GRU is close to the input observation. Attention block near 0 means that the effect of past behaviors is also negligible.
![image](https://user-images.githubusercontent.com/41103541/160962182-cecefde6-0466-4274-b996-6f14426af69b.png)
The content between the brackets will make it more clear, i will explain better the doc